modules/objdetect/include/opencv2/objdetect/face.hpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef _OPENCV_DNN_FACE_CORE_HPP_
+#define _OPENCV_DNN_FACE_CORE_HPP_
+
+#include <opencv2/core.hpp>
+
+/** @defgroup dnn_face DNN-based face detection and recognition
+ */
+
+namespace cv
+{
+
+/** @brief DNN-based face detector, model download link: https://github.com/ShiqiYu/libfacedetection.train/tree/master/tasks/task1/onnx.
+ */
+class CV_EXPORTS_W DNNFaceDetector
+{
+public:
+    virtual ~DNNFaceDetector() {};
+
+    /** @brief A simple interface to detect face from given image
+     *
+     *  @param image an image to detect
+     *  @param faces detection results stored in a cv::Mat
+     */
+    CV_WRAP virtual int detect(InputArray image, OutputArray faces) = 0;
```

Comment:
I think, we need some standard, specification, which landmarks represent what. Perhaps, the first 5 landmarks should be the eyes (in the specified order, e.g. left, then right), nose and the mouth left's and right's point, then all the other, more detailed landmarks (if any).

Reply:
Currently I defined the landmarks as the order you said. There are several standards in term of facial landmarks, such as 5-landmark definition that we used, [68-landmark definition](https://ibug.doc.ic.ac.uk/resources/300-W/), [98-landmark definition](https://wywu.github.io/projects/LAB/WFLW.html), [21-landmark definition](https://www.tugraz.at/institute/icg/research/team-bischof/lrs/downloads/aflw/) and maybe more. The order of landmarks vary from different definitions. We need a serious discussion on this issue.

Reply:
Currently I defined the landmarks as the order you said. There are several standards in term of facial landmarks, such as 5-landmark definition that we used, [68-landmark definition](https://ibug.doc.ic.ac.uk/resources/300-W/), [98-landmark definition](https://wywu.github.io/projects/LAB/WFLW.html), [21-landmark definition](https://www.tugraz.at/institute/icg/research/team-bischof/lrs/downloads/aflw/) and maybe more. The order of landmarks vary from different definitions. We need a serious discussion on this issue.

Reply:
Currently use 5-landmark definition as standard, but can be extended whenever needed.